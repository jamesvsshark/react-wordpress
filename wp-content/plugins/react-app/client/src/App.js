import React from "react";
import ReactTable from "react-table";
import Logo from "./logo.svg";
import 'react-table/react-table.css'

export class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { liked: false };
  }

  render() {
    // if (this.state.liked) {
    //   return ;
    // }

    // return <button onClick={() => this.setState({ liked: true })}>Like</button>;

    const data = [{
      name: 'Tanner Linsley',
      age: 26,
      friend: {
        name: 'Jason Maurer',
        age: 23,
      }
    }]
  
    const columns = [{
      Header: 'Name',
      accessor: 'name' // String-based value accessors!
    }, {
      Header: 'Age',
      accessor: 'age',
      Cell: props => <span className='number'>{props.value}</span> // Custom cell components!
    }, {
      id: 'friendName', // Required because our accessor is not a string
      Header: 'Friend Name',
      accessor: d => d.friend.name // Custom value accessors!
    }, {
      Header: props => <span>Friend Age</span>, // Custom header components!
      accessor: 'friend.age'
    }]

    return (
    <>
      <img src={Logo} />
      <ReactTable data={data} columns={columns}/>
    </>
    )
  }
}

export default App;
