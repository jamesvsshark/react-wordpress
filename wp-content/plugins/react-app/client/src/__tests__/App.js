import {App} from '../App';

describe("Sample app", () => {
  it("should always be true", () => {
    expect(true).toBe(true);
  });

  it('should be defined and match snapshot', () => {
      expect(App).toBeDefined();
      expect(App).toMatchSnapshot();
  })
});
