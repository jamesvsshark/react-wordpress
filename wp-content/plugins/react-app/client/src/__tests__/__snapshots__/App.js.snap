// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample app should be defined and match snapshot 1`] = `[Function]`;
